lib/mongo/client.rb

```diff
@@ -1103,6 +1103,18 @@ def cluster_modifying?(new_options)
     # but does not check for interactions between combinations of options.
     def validate_new_options!(opts = Options::Redacted.new)
       return Options::Redacted.new unless opts
+      if opts[:read_concern]
+        given_keys = opts[:read_concern].keys
+        allowed_keys = ['level']
+        # raise an error for non user-settable options
+        if opts[:read_concern][:after_cluster_time]
+          raise Mongo::Error::OperationFailure.new('after_cluster_time is not a user-settable option')
```

Comment:
This needs to be a different exception class since OperationFailure is for errors produced by the server. I am not seeing one that we have that would fit, perhaps something like InvalidOption could be appropriate.